### DISCUSSION TITLE
Proper way to do conditional statements with new async tp.file.exists?

### DISCUSSION THREAD

**Discussion by Zebrazilla** [SilentVoid13/Templater#877] [Help] [answered]
After updating I am having trouble with my conditional include depending on if a file exists or not. Prior to the change of tp.file.exists from sync to async this worked well:
`<%-* if(tp.file.exists(tp.date.now("MM-DD", 0, tp.file.title))) { tR += (await tp.file.include("[[" + tp.date.now("MM-DD", 0, tp.file.title) + "]]")).split('---')[2]; } -%>`
In short, this looks for a file formatted as current MM-DD date. If it exists, include it.

Now that tp.file.exists is async, it works whenever such a file exists, but whenever such a file doesn't exist it fails completely and the template breaks: 
![image](https://user-images.githubusercontent.com/6796156/197360162-38136e06-6467-4c39-af39-d3cb1af5e254.png)

I obviously need to adapt the code to work aynsc, but my knowledge of how to do so is limited. I've tried simply making the conditional contain await, like so:
`if(await tp.file.exists ...`
but that didn't work, it seems to always return false?

Any ideas on how I could do this instead? Help is greatly appreciated!

**Comment by AB1908** [comment, marked as answer]
Separate it out like so:

```
const fileExists = await tp.file.exists();
if (fileExists)
```

**Reply by Zebrazilla** [reply]
Thanks, that works! I also needed to adjust filename to include full path, once that was in place it all works properly again.
